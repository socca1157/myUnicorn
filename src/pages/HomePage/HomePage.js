import React, { Component } from "react";
import {
    View,
    Text,
    AsyncStorage,
    NativeModules,
    Platform
} from "react-native";
import { Button } from "react-native-elements";

import AudioPlayer from "../../components/AudioPlayer/";
import styles from "../../components/Theme/Theme";

class HomePage extends Component {
    static navigationOptions = {
        title: "myUnicorn"
    };

    constructor() {
        super();
        this.logout = this.logout.bind(this);
        this.AudioPlayerRef = React.createRef();
        this.state = {
            userName: ""
        };
        this.getUserName();
    }
    async getUserName() {
        const userName = await AsyncStorage.getItem("userName");
        this.setState({ userName: userName });
    }

    async logout() {
        await this.AudioPlayerRef.current.stopMusic(); // make sure music is stopped
        NativeModules.NativeAlert.openAlert("Success!", "You've been logged out."); // our super fancy native bridge
        await AsyncStorage.setItem("lastUser", ""); // using this to Logout instead of .clear(), so we keep the displayName until next register
        this.props.navigation.navigate("ReAuth"); // now send back to login screen
    }

    render() {
        return (
            <View style={styles.container}>
                <Text style={styles.welcome}>
                    Welcome {this.state.userName} toðŸ¦„ paradise!
                </Text>
                <AudioPlayer
                    ref={this.AudioPlayerRef}
                    url="https://austincameron.com/Pink%20Fluffy%20Unicorns%20Dancing%20on%20Rainbows%20-%20Fluffle%20Puff%20.mp3"
                    looping={Platform.OS === "ios" ? true : false}
                />
                <Button
                    onPress={this.logout}
                    type="outline"
                    title="Logout"
                    style={{ marginTop: 50 }}
                />
            </View>
        );
    }
}
export default HomePage;