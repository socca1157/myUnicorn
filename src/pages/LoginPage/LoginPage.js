import React, { Component } from "react";
import {
    Platform,
    StyleSheet,
    Text,
    View,
    Alert,
    AsyncStorage
} from "react-native";

import { Input, Button } from "react-native-elements";
import Icon from "react-native-vector-icons/FontAwesome";

import styles from "../../components/Theme/Theme";
import { firebaseLogin } from "../../api/Firebase.js";

class LoginPage extends Component {
  static navigationOptions = {
    title: "Welcome Back"
  };
  constructor() {
    super();
    this.state = {
      formVerify: false,
      formError: false,
      responseError: null,
      isLoggingIn: false,
      email: "",
      password: ""
    };
    this.formVerify = this.formVerify.bind(this);
  }

  formVerify = async nextStep => { // check the 2 required to enable button
    if (
        /(\S+@\S+\.\S+)/.test(this.state.email) && // possibly implement RFC-2822 later on
        this.state.password.length > 5 // 6 or higher
    ) {
      this.setState({ formVerify: true });
      this.setState({ formError: false }); // reset error
    } else {
      this.setState({ formVerify: false }); // just in case inputs change later
    }
    if (nextStep && this.state.formVerify) {
      // they've pressed the Login button
      // now check creds against local db

      this.setState({ isLoggingIn: true });
      firebaseLogin(this.state.email, this.state.password).then(async user => {
        console.log(user);
        if (user.code) {
          console.log("made it here");
          let errorMsg;
          switch (user.code) {
            case "auth/user-not-found":
            case "auth/wrong-password":
              errorMsg = "Invalid email address or password";
              break;
            case "auth/email-already-in-use":
              errorMsg = "This email address is already taken.";
              break;
            case "auth/invalid-email":
              errorMsg = "Invalid e-mail address format."; // shouldn't get this, but just in case.
              break;
            case "auth/weak-password":
              errorMsg = "Password is too weak."; // shouldn't get this, but just in case.
              break;
            default:
              errorMsg = "Check your internet connection.";
          }
          this.setState({ responseError: "Oops!\r\n" + errorMsg }); // let user know what went wrong
          this.setState({ isLoggingIn: false }); // re-activate button
        } else {
          // now let them in
          await AsyncStorage.setItem("lastUser", this.state.email); // enable auto login
          this.props.navigation.navigate("App");
        }
      });
    }
  };

  render() {
    return (
        <>
        <View style={styles.container}>
          <Text style={styles.welcome}>Log in to MyUnicorn!</Text>
          <Text style={styles.tagline}>
            The #1 Resource for Pink Fluffy Unicorns Dancing on Rainbows.
          </Text>

          <Input
              leftIcon={
              <Icon
                name="user-o"
                size={18}
                color="#222"
                style={{ marginRight: 15 }}
              />
            }
              placeholder="Email Address"
              value={this.state.email}
              onChangeText={email => this.setState({ email }, this.formVerify)}
          />
          <Input
              leftIcon={
              <Icon
                name="unlock-alt"
                size={18}
                color="#222"
                style={{ marginRight: 15 }}
              />
            }
              placeholder="Password"
              value={this.state.password}
              secureTextEntry={true}
              onChangeText={password =>
              this.setState({ password }, this.formVerify)
            }
          />
          <Text style={styles.errorMsg}>
            {this.state.formError
                ? "Oops! Please double check your details."
                : ""}
          </Text>
          <View style={styles.buttonContainer}>
            <Text style={styles.errorMsg}>{this.state.responseError}</Text>
            <Button
                onPress={() => this.formVerify("nextStep")}
                disabled={!this.state.formVerify || this.state.isLoggingIn}
                loading={this.state.isLoggingIn}
                title="Log In"
                type="solid"
                raised={true}
                icon={
                <Icon name="sign-in" size={15} style={{ marginRight: 10 }} />
              }
            />
            <Text style={styles.or}>or</Text>
            <Button
                onPress={() => this.props.navigation.navigate("SignUp")}
                title="Register"
                type="clear"
            />
          </View>
        </View>
        </>
    );
  }
}

export default LoginPage;
