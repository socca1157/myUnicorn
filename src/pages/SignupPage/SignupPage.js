import React, { Component } from "react";
import { Platform, Text, View, Alert, AsyncStorage } from "react-native";

import { Input, Button } from "react-native-elements";
import Icon from "react-native-vector-icons/FontAwesome";

import styles from "../../components/Theme/Theme";
import { firebaseCreateAccount } from "../../api/Firebase.js";

class SignupPage extends Component {
    static navigationOptions = {
        title: "myUnicorn"
    };
    constructor() {
        super();
        this.state = {
            formVerify: false,
            isCreatingAccount: false,
            responseError: null,
            email: "",
            password: "",
            name: ""
        };
        this.formVerify = this.formVerify.bind(this);
    }

    formVerify = async nextStep => { // check the 3 required to enable button
        if (
            /(\S+@\S+\.\S+)/.test(this.state.email) && // possibly implement RFC-2822 later on
            this.state.password.length > 5 && // 6 or higher
            this.state.name &&
            /^([a-zA-Z\s]*)$/.test(this.state.name) // only letters and spaces
        ) {
            this.setState({ formVerify: true });
        } else {
            this.setState({ formVerify: false }); // just in case inputs change later
        }
        if (nextStep && this.state.formVerify) {
            // they've pressed the Signup button and it's validated

            // first, lets check if user already exists
            try {
                let value = await AsyncStorage.getItem("userName-" + this.state.email);

                if (value != null) {
                    //user found, login instead
                    Alert.alert(
                        "Email Already in Use",
                        "Would you like to log in instead?",
                        [
                            {
                                text: "Ok",
                                onPress: () => this.props.navigation.navigate("LogIn")
                            },
                            {
                                text: "Cancel",
                                onPress: () => console.log("Cancel Pressed"),
                                style: "cancel"
                            }
                        ],
                        { cancelable: false }
                    );
                } else {
                    //no users found, so we save.

                    this.setState({ isCreatingAccount: true }); // prevent multiple button presses

                    firebaseCreateAccount(
                        this.state.name,
                        this.state.email,
                        this.state.password
                    ).then(async result => {
                        if (result.code) {
                            let errorMsg;
                            switch (result.code) {
                                case "auth/email-already-in-use":
                                    errorMsg = "This email address is already taken.";
                                    break;
                                case "auth/invalid-email":
                                    errorMsg = "Invalid e-mail address format."; // shouldn't get this, but just in case.
                                    break;
                                case "auth/weak-password":
                                    errorMsg = "Password is too weak."; // shouldn't get this, but just in case.
                                    break;
                                default:
                                    errorMsg = "Check your internet connection.";
                            }
                            this.setState({ responseError: "Oops!\r\n" + errorMsg }); // let user know what went wrong
                            this.setState({ isCreatingAccount: false }); // re-activate button
                        } else {
                            // now let them in
                            await AsyncStorage.setItem("lastUser", this.state.email); // enable auto login
                            await AsyncStorage.setItem("userName", this.state.name); // Save name, to be polite.
                            this.props.navigation.navigate("App");
                        }
                    });
                }
            } catch (error) {
                // Error retrieving data
                // console.log(error);
            }
        }
    };

    render() {
        return (
            <>
            <View style={styles.container}>
                <Text style={styles.welcome}>Welcome to MyUnicorn!</Text>
                <Text style={styles.tagline}>
                    The #1 Resource for Pink Fluffy Unicorns Dancing on Rainbows.
                </Text>

                <Input
                    leftIcon={
              <Icon
                name="user-o"
                size={18}
                color="#222"
                style={{ marginRight: 15 }}
              />
            }
                    placeholder="Email Address"
                    value={this.state.email}
                    keyboardType="email-address"
                    onChangeText={email => this.setState({ email }, this.formVerify)}
                />
                <Input
                    leftIcon={
              <Icon
                name="lock"
                size={18}
                color="#222"
                style={{ marginRight: 15 }}
              />
            }
                    placeholder="Password"
                    value={this.state.password}
                    secureTextEntry={true}
                    onChangeText={password =>
              this.setState({ password }, this.formVerify)
            }
                />
                <Input
                    leftIcon={
              <Icon
                name="address-card-o"
                size={18}
                color="#222"
                style={{ marginRight: 15 }}
              />
            }
                    placeholder="Full Name"
                    value={this.state.name}
                    onChangeText={name => this.setState({ name }, this.formVerify)}
                />
                <Text style={styles.errorMsg}>{this.state.responseError}</Text>

                <View style={styles.buttonContainer}>
                    <Button
                        onPress={() => this.formVerify("nextStep")}
                        disabled={!this.state.formVerify || this.state.isCreatingAccount}
                        loading={this.state.isCreatingAccount}
                        title="Register"
                        type="solid"
                        raised={true}
                        icon={
                <Icon
                  name="sign-in"
                  size={15}
                  color="#fff"
                  style={{ marginRight: 10 }}
                />
              }
                    />
                    <Text style={styles.or}>or</Text>
                    <Button
                        onPress={() => this.props.navigation.navigate("LogIn")}
                        title="Log In"
                        type="clear"
                    />
                </View>
            </View>
            </>
        );
    }
}

export default SignupPage;