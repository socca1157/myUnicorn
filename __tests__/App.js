/**
 * @format
 * @lint-ignore-every XPLATJSCOPYRIGHT1
 */

import "react-native";
import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

/**
 *  Testing our pages
 */
import SignupPage from "../src/pages/SignupPage/SignupPage";
describe("SignupPage", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<SignupPage />);
    expect(wrapper).toMatchSnapshot();
  });
});

import LoginPage from "../src/pages/LoginPage/LoginPage";
describe("LoginPage", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<SignupPage />);
    expect(wrapper).toMatchSnapshot();
  });
});

import HomePage from "../src/pages/HomePage/HomePage";
describe("HomePage", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<SignupPage />);
    expect(wrapper).toMatchSnapshot();
  });
});
