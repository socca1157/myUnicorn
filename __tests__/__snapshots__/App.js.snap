// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage renders correctly 1`] = `ShallowWrapper {}`;

exports[`LoginPage renders correctly 1`] = `ShallowWrapper {}`;

exports[`SignupPage renders correctly 1`] = `ShallowWrapper {}`;
